flytekit/types/file/file.py

```diff
@@ -184,6 +209,12 @@ def remote_source(self) -> str:
         """
         return self._remote_source
 
+    def trigger_download(self):
```

Comment:
Should we use this in __fspath__ instead of self._downloader()?

```
def __fspath__(self):
        # This is where a delayed downloading of the file will happen
        if not self._downloaded:
            self._downloader()
            self._downloaded = True
        return self._path
```

Also I think we need to add the `download cache` check here too!